assembly/runtime/promise.ts

```diff
@@ -0,0 +1,145 @@
+import { util } from "./util";
+import { env } from "./env";
+import { u128 } from ".";
+
+function encodeArgs<T>(t: T) : Uint8Array {
+  if (t instanceof Uint8Array) {
+    return t;
+  }
+  return encode<T>(t);
+}
+
+// TODO: not sure if these types make any sense at all
+export type Balance = u128;
+export type Gas = u64;
+export type PublicKey = Uint8Array;
+export type AccountId = Uint8Array;
+
+export type CreateAccount = "CreateAccount"; // TODO: this seems wierd
+export type DeployContract = { code: Uint8Array };
+export type FunctionCall = { methodName: Uint8Array, args: Uint8Array, amount: Balance, gas: Gas };
+export type Transfer = { amount: Balance };
+export type Stake = { amount: Balance, publicKey: PublicKey };
+export type AddFullAccessKey = { publicKey: PublicKey };
+export type AddAccessKey = { publicKey: PublicKey, allowance: Balance, receiverId: AccountId, methodNames: Uint8Array[] };
+export type DeleteKey = { publicKey: PublicKey };
+export type DeleteAccount = { beneficiaryId: AccountId };
+
+export type Action =
+  | CreateAccount
+  | DeployContract
+  | FunctionCall
+  | Transfer
+  | Stake
+  | AddFullAccessKey
+  | AddAccessKey
+  | DeleteKey
+  | DeleteAccount;
+
+/**
+ * Chain actions
+ *
+ * let p1 = new ActionPromise("bob_near").create_account();
+ * let p2 = new ActionPromise("alice_near").create_account().transfer(100);
+ */
+export class ContractPromiseBatch {
+  actions: Action[];
+  after: ContractPromiseBatch;
```

Comment:
Should this be set in the `then` method?

Reply:
Do we need it?  Same with actions above?

Reply:
not clear, removed both of these